syntax = "proto3";

// Request header. Immediately followed by an (encrypted) Rq message of length
// payload_length.
//
message RqHeader {
    fixed32 partner_id  = 1;
    bytes iv = 2;
    fixed32 payload_length = 3;
}

// Response header. Immediately followed by an (encrypted) Rs message of length
// payload_length.
//
message RsHeader {
    bytes iv=1;
    fixed32 payload_length=2;
}

message Rq {
    uint32 sw_version = 3;
    int64 client_mac = 4;
    Aps aps = 5;
    LteCells lte_cells = 6;
    uint32 checksum = 7;
}

message Aps {
    enum ApBand {
        UNKNOWN = 0;
        BAND_2_4 = 1;
        BAND_5 = 2;
    }

    repeated int64 mac = 1;
    repeated sint32 rssi = 2;
    repeated ApBand band = 3;
    repeated bool connected = 4;
}

message LteCells {
    repeated uint32 age = 1;
    repeated uint32 eucid = 2;
    repeated uint32 mcc = 3;
    repeated uint32 mnc = 4;
    repeated sint32 rssi = 5;
}
